libs/subliminal_patch/providers/whisperai.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import
+import logging
+
+from requests import Session
+
+from subliminal_patch.subtitle import Subtitle
+from subliminal_patch.providers import Provider
+from subliminal import __short_version__
+from subliminal.exceptions import ConfigurationError
+from subzero.language import Language
+from subliminal.video import Episode, Movie
+
+import ffmpeg
+
+
+logger = logging.getLogger(__name__)
+
+
+class WhisperAISubtitle(Subtitle):
+    '''Whisper AI Subtitle.'''
+    provider_name = 'whisperai'
+    hash_verifiable = False
+
+    def __init__(self, language, video):
+        super(WhisperAISubtitle, self).__init__(language)
+
+        self.video = video
+
+    @property
+    def id(self):
+        return self.video.original_name
+
+    def get_matches(self, video):
+        matches = set()
+
+        if isinstance(video, Episode):
+            matches.update(["series", "season", "episode"])
+        elif isinstance(video, Movie):
+            matches.update(["title"])
+
+        return matches
+
+
+class WhisperAIProvider(Provider):
+    '''Whisper AI Provider.'''
+
+    # These are all the languages Whisper supports.
+    # from whisper.tokenizer import LANGUAGES
+    whisper_languages = ['eng', 'chi', 'ger', 'spa', 'rus', 'kor', 'fre', 'jpn', 'por', 'tur', 'pol', 'cat', 'dut', 'ara', 'swe', 'ita', 'ind', 'hin', 'fin', 'vie', 'heb', 'ukr', 'gre', 'may', 'cze', 'rum', 'dan', 'hun', 'tam', 'nor', 'tha', 'urd', 'hrv', 'bul', 'lit', 'lat', 'mao', 'mal', 'wel', 'slo', 'tel', 'per', 'lav', 'ben', 'srp', 'aze', 'slv', 'kan', 'est', 'mac', 'bre', 'baq', 'ice', 'arm', 'nep', 'mon', 'bos', 'kaz', 'alb', 'swa', 'glg', 'mar', 'pan', 'sin', 'khm', 'sna', 'yor', 'som', 'afr', 'oci', 'geo', 'bel', 'tgk', 'snd', 'guj', 'amh', 'yid', 'lao', 'uzb', 'fao', 'hat', 'pus', 'tuk', 'nno', 'mlt', 'san', 'ltz', 'bur', 'tib', 'tgl', 'mlg', 'asm', 'tat', 'haw', 'lin', 'hau', 'bak', 'jav', 'sun']
+
+    languages = {
+        Language.fromalpha3b(lang) for lang in whisper_languages
+    }
+    languages.update(set(Language.rebuild(lang, hi=True) for lang in languages))
```

Comment:
Is this applicable? should i leave it in here? @vitiko98

Reply:
the hi=True and forced=True part specifically is what i mean.

Reply:
I think we should remove it in a future update. Whisper is trying to build twice when forced subtitles are set up, but they are the same than non forced subtitles.

Reply:
That makes sense. At the time I didn't understand much about providers so I was copying it as closely as I could without understanding how it worked.

Reply:
I would have done the same. I just digged into it, because I automated the WhisperAI generation in case nothing else is found by lowering the scores. It started to create a lot of forced subtitles, which were only normal subtitles.
Are you making a new pull request to fix it or someone else should do it?

Reply:
I don't have a lot of free time, so it would be at least a few days before I could push out a fix. Please feel free to make a PR! Thank you.